File: JavaScript/901-Online Stock Span.js

var StockSpanner = function() {
    this.stack = [];
};

/** 
 * @param {number} price
 * @return {number}
 */
StockSpanner.prototype.next = function(price) {
    let weight = 1;
    while (this.stack.length > 0 && price >= this.stack[this.stack.length-1][0]){
        weight += this.stack.pop()[1];
    }
    this.stack.push([price, weight]);
    return weight;
};

/** 
 * Your StockSpanner object will be instantiated and called as such:
 * var obj = new StockSpanner()
 * var param_1 = obj.next(price)
 */

File: 901-Online Stock Span.py
class StockSpanner(object):

    def __init__(self):
        self.stack = []

    def next(self, price):
        """
        :type price: int
        :rtype: int
        """
        weight = 1
        while self.stack and price >= self.stack[-1][0]:
            weight += self.stack.pop()[1]
        self.stack.append((price, weight))
        return weight

# Your StockSpanner object will be instantiated and called as such:
# obj = StockSpanner()
# param_1 = obj.next(price)